### PR TITLE
autodetect backend host when running in dev mode

### PR DESCRIPTION
**Pull request by chrisdevereux**
This PR does 2 things:

1) Centralises config settings into a single module
2) When running in dev mode, infers the backend url from the react native packager URL

Should make collaboration easier and will also be useful when we need to split our staging and production deployments

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,6 +63,8 @@
     "react-navigation-header-buttons": "^2.1.2",
     "subscriptions-transport-ws": "^0.9.15",
     "tslib": "^1.9.3",
+    "typesafe-actions": "^3.0.0",
```
Comment (brortao) [inline package.json]: do we need this dependency?

**Comment by brortao** [comment]
this is great! i've been thinking that something like this would be nice but didn't realize it was possible.

any idea why the build is failing? it's missing the `ENGINE_API_KEY` environment variable, but i just verified that it is set in travis.

File: package.json (hunk 1 above)
Comment (chrisdevereux) [reply package.json]: Ahhh, think I must have hit 'save' after switching branches from another one it gets added in

**Comment by chrisdevereux** [comment]
Yeah, Travis doesn't expose environmental variables to PRs from forks. Which makes a lot of sense, as otherwise someone could open a PR, then steal secrets from the environment.

Once #114 is merged, this should pass

**Pull request merged**